cask/src-0/cask/main/Routes.scala

```diff
@@ -0,0 +1,21 @@
+package cask.main
+
+import cask.router.RoutesEndpointsMetadata
+
+import language.experimental.macros
+
+trait Routes{
+
+  def decorators = Seq.empty[cask.router.Decorator[_, _, _]]
+  private[this] var metadata0: RoutesEndpointsMetadata[this.type] = null
+  def caskMetadata =
+    if (metadata0 != null) metadata0
+    else throw new Exception("Routes not yet initialized")
+
+  protected[this] inline def initialize(): Unit = ${
+    RoutesEndpointsMetadata.setRoutesImpl[this.type](
+      '{(x: RoutesEndpointsMetadata[this.type]) => metadata0 = x}
+    )
```

Comment:
FYI: we can't use the typeclass approach in dotty, because on failure the "nice" error messages from the macros are ignored, and instead the generic "implicit not found" message is displayed. Hence we use this trick to pass in the setter as a closure, so that we can use a regular macro.